inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -39,68 +59,94 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
-                } else if (cmp(nearest, "simple")) {
+                } else if (cmp(near, simple)) {
                     nearestMode = InterpolateNearestMode::Simple;
+                } else {
+                    VPU_THROW_EXCEPTION << "Current Interpolate does not support this nearest mode; layer name = " << _layer->name;
                 }
 
                 _stageBuilder->addResampleNearestStage(model,
                                                        _layer->name,
                                                        _layer,
-                                                       antialias,
+                                                       anti,
                                                        coordinateTransformationMode,
                                                        nearestMode,
                                                        -1.0f,
                                                        input,
                                                        output);
-            } else if (cmp(interpolateMode, "linear") || cmp(interpolateMode, "linear_onnx")) {
+            } else if (cmp(interpolateMode, linear) || cmp(interpolateMode, linear_onnx)) {
                 // current "Interp" supports modes "align_corners" and "asymmetric" only
                 // other "Interpolate" modes are translated to the default ones
-                const auto coordinateTransformationMode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
+                const auto coordinateTransformation = _layer->GetParamAsString(coordinate_transformation_mode, half_pixel);
+                InterpolateCoordTransMode coordinateTransformationMode = InterpolateCoordTransMode::HalfPixel;
+                InterpolateMode mode = InterpolateMode::Linear;
+
+                if (cmp(interpolateMode, linear_onnx)) {
+                    mode = InterpolateMode::LinearOnnx;
+                }
+
+                if (cmp(coordinateTransformation, asymmetric)) {
+                    coordinateTransformationMode = InterpolateCoordTransMode::Asymmetric;
+                } else if (cmp(coordinateTransformation, half_pixel)) {
+                    coordinateTransformationMode = InterpolateCoordTransMode::HalfPixel;
+                } else if (cmp(coordinateTransformation, pytorch_half_pixel)) {
+                    coordinateTransformationMode = InterpolateCoordTransMode::PytorchHalfPixel;
+                } else if (cmp(coordinateTransformation, tf_half_pixel_for_nn)) {
+                    coordinateTransformationMode = InterpolateCoordTransMode::TfHalfPixelForNn;
+                } else if (cmp(coordinateTransformation, align_corners)) {
+                    coordinateTransformationMode = InterpolateCoordTransMode::AlignCorners;
+                } else {
+                    VPU_THROW_EXCEPTION << "Current Interpolate does not support this coordinate transformation mode; layer name = " << _layer->name;
```

Comment:
Nitpick: I think VPU_THROW_FORMAT should be used instead

Reply:
done